modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -157,7 +158,24 @@ func (h *Handler) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 		}
 		commonScheme = scheme
 
-		h.Upstreams = append(h.Upstreams, &Upstream{Dial: dialAddr})
+		parsedAddr, err := caddy.ParseNetworkAddress(dialAddr)
+		if err != nil {
+			return d.WrapErr(err)
+		}
+
+		if parsedAddr.StartPort == 0 && parsedAddr.EndPort == 0 {
```

Comment:
Why 0? :thinking:

Reply:
I don't remember 🤔 might've been for placeholders which have no port (assuming the port is part of the placeholder value) but I'll need to take a closer look to remind myself. I'll add comments.

Reply:
Oh right, unix networks have port 0